git.md

```diff
@@ -1,2 +1,130 @@
+* apply to shared commits (maybe after cleaning commits locally).
+
+## tl;dr
+
+* Shared commits should not be changed.
+* Changes in each shared commit should be closely related.
+
+## 1. Preserve shared history
+
+Shared commits should never change. Some Git commands preserve the history, but
+others don't. For example, `git commit --amend` and `git rebase --interactive`
+alter the git history; they help to overwrite  a messy commit history with
+a cleaner one; you should use them only before sharing your work. To cleanup
+shared history use instead `git revert`.  If you try to push altered history by
+accident, Git will stop you; to benefit from this protection  avoid `--force`
+(or `-f`) when pushing to shared repositories.
+
+## 2. Keep commits focused
+
+A commit should focus on related changes only. For example, fixing two
+different bugs should produce two separate commits. Small commits make it
+easier for other developers to understand the changes and roll them back if
+something went wrong. You can avoid sharing unwanted changes via .gitignore. If
+you can't succinctly describe what the commit is doing in a couple of words,
+then it is likely too big.
+
+See also:
+
+* [The right size of a PR](https://github.com/2DegreesInvesting/practices/discussions/3).
+* [.gitignore templates](https://github.com/github/gitignore).
+
+TODO: Undoing things with Git
+
+## 3. Commit frequently
+
+Committing frequently keeps your commits small. Small commits are more likely
+to be focused; they are easier to review, and easier to merge because they
+offer fewer opportunities for conflicts.
+
+## 4. The work you share should be complete
+
+Each commit and feature should do what it promises, completely; and reverting
+it should undo what that commit or feature meant to do, also completely. If
+the goal of one commit requires changes in two files, that one commit should
+include changes to both files. Similarly, if the goal of a feature requires
+two kinds of changes, that feature should likely include two commits.
+
+ TODO: Undoing things with Git
+
+## 5. Explain how to test the effect of the changes you made
+
+Testing your code is important, particularly when sharing it with
+others. Automated tests are the best but not the only way to test the effects
+of the changes you made. However you did it, explain how you ensured that the
+code behaves as you expect, and how someone else may reproduce your results.
+
+## 6. Structure each commit message like an email
+
+Like an email, commit messages have a subject and an optional body. The subject
+should start with a line summarizing your changes (aim for 50 characters or
+less). Use the imperative, present tense -- for example, "Fix typo",
+not "Fixed typo" or "Fixes typo". The message body, if present,
+should follow a blank line; wrap around 70 characters; and explain the
+motivation for the change ("why", not "what" or "how").
```

Comment:
I didn't know that the focus of the body is why, and not what or how! That's cool.

Reply:
Yeah, that's cool. I guess the reason is that the intent is the only thing that the changes themselves don't capture, so explaining "why" is most important. Summarizing "what" or "how" might still be nice, but the effortful reader can get it from the change history.